clang/lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -247,6 +247,20 @@ class StmtComparer {
 
   bool IsStmtEquivalent(const GenericSelectionExpr *E1,
                         const GenericSelectionExpr *E2) {
+    if (!IsStructurallyEquivalent(Context,
```

Comment:
Could you please add comments to the new checks that explains why they are needed?

I don't doubt that they are important, but I haven't touched this code in a very long time, and I'm sure that other readers of this code would appreciate it.